lib/types.nix

```
@@ -674,15 +674,15 @@ rec {
     # A module to be imported in some other part of the configuration.
     # `staticModules`' options will be added to the documentation, unlike
     # options declared via `config`.
-    deferredModuleWith = attrs@{ staticModules ? [] }: mkOptionType {
+    deferredModuleWith = attrs@{ staticModules ? [], specialArgs ? {} }: mkOptionType {
```

These `specialArgs` could be confused for the "normal" `specialArgs` in `evalModules` and `submoduleWith`.
Fundamentally, that's not something that can be supported without doing fixpoint _iteration_ - invoking modules multiple times.
So we should make clear that this is only for documentation. One way to do this is to bundle documentation parameters into its own attribute:

```suggestion
    deferredModuleWith = attrs@{ staticModules ? [], docs ? {} }: mkOptionType {
```

This would be overkill for a single parameter, but another possible documentation parameter is `docs.modules`, which would also only be added while generating docs.

- `config`: config only
- `options`/`staticModules`: config and docs
- `options`/`docs.modules`: docs only

`docs.modules` does not necessarily have to be implemented in this PR.